extensions/patterns/configure_ec2/template_rhdh/create_ec2_instance.yml

```diff
@@ -0,0 +1,185 @@
+      description: Create EC2 instance in the Amazon cloud
+      required:
+        - token
+        - inventory
+      properties:
+        token:
+          title: Token
+          type: string
+          description: Oauth2 token
+          ui:field: AAPTokenField
+          ui:widget: password
+          ui:backstage:
+            review:
+              show: false
+        inventory:
+          title: Inventory
+          description: Please enter the inventory you want to use the services on
+          resource: inventories
+          ui:field: AAPResourcePicker
+        credentials:
+          title: Credentials
+          description: Select credentials for accessing the nodes this job will be ran
+            against. You can only select one credential of each type. For
+            machine credentials (SSH), checking "Prompt on launch" without
+            selecting credentials will require you to select a machine
+            credential at run time. If you select credentials and check "Prompt
+            on launch", the selected credential(s) become the defaults that can
+            be updated at run time.
+          type: array
+          ui:field: AAPResourcePicker
+          resource: credentials
+        verbosity:
+          title: Verbosity
+          description: Control the level of output Ansible will produce as the playbook executes.
+          resource: verbosity
+          ui:field: AAPResourcePicker
+    - title: Survey
+      required:
+        - aws_region
+        - instance_name
+        - instance_type
+      description: Create EC2 instance in the Amazon cloud
+      properties:
+        aws_region:
+          title: AWS Region
+          description: AWS region where resources should be created
+          type: string
+          ui:options:
+            rows: 5
+        instance_name:
+          title: Instance Name
+          description: Name of EC2 instance to create
+          type: string
+          ui:options:
+            rows: 5
+        instance_type:
+          title: Instance Type
+          description: Type of EC2 instance to create (e.g., t2.micro, m5.large)
+          type: string
+          ui:options:
+            rows: 5
+        ami_id:
+          title: AMI ID
+          description: Amazon Machine Image (AMI) ID to use for the instance, if not provided will default to the RHEL 9 AMI for the provided region and instance type
+          type: string
+          ui:options:
+            rows: 5
+        key_name:
+          title: Key Pair Name
+          description: Name of key pair to use or create for SSH access to the EC2 instance. Defaults to '{{ instance_name }}-key'
+          type: string
+          ui:options:
+            rows: 5
+        wait_for_state:
+          title: Wait for State
+          description: Whether to wait for the EC2 instance to be in the running state before continuing. Defaults to true
+          type: string
+          enum:
+            - ""
```

Comment:
Do we need to provide an empty string option here?

Reply:
A guess: '' will let playbook select default value. We just do not want user to type random string, hence enum.